crates/router/src/services/authentication/detached.rs

```diff
@@ -61,13 +64,25 @@ impl ExtractedPayload {
                 message: format!("`{}` header not present", HEADER_AUTH_TYPE),
             })?;
 
+        let key_id = headers
+            .get(HEADER_KEY_ID)
+            .and_then(|value| value.to_str().ok())
+            .ok_or_else(|| ApiErrorResponse::InvalidRequestData {
+                message: format!("`{}` header is invalid or not present", HEADER_KEY_ID),
+            })
+            .map_err(error_stack::Report::from)
```

Comment:
This shouldn't error out. In case where the `key_id` is not present, it should instead set it to `None`